File: project_euler/Problem091/Problem091.py
class Problem091:
    def solution(self) -> None:
        target = 50
        result = target * target * 3
        for i in range(1, target):
            for j in range(1, target + 1):
                divisor = self.__common_divisor(i, j)
                result += min((target - i) * divisor // j, j * divisor // i) << 1
        print(result)

    def __common_divisor(self, i: int, j: int) -> int:
        while i % j:
            i, j = j, i % j
        return j

File: project_euler/Problem091/Problem091.java
package project_euler;

public class Problem091 {
    public void solution() {
        int target = 50;
        int result = target * target * 3;
        for (int i = 1; i < target; ++i) {
            for (int j = 1; j <= target; ++j) {
                int divisor = commonDivisor(i, j);
                result += Math.min((target - i) * divisor / j, j * divisor / i) << 1;
            }
        }
        System.out.println(result);
    }

    private int commonDivisor(int i, int j) {
        while (0 != i % j) {
            int auxiliary = i % j;
            i = j;
            j = auxiliary;
        }
        return j;
    }
}
